tests/test_documents.py

```diff
@@ -47,6 +62,34 @@ async def doc_fixture(client) -> DocFixture:
     )
 
 
+async def upload_test_pdf(client, document_id, payload):
+    document = await upload_document_content.asyncio(
+        client=client,
+        document_id=document_id,
+        body=BodyUploadDocumentContent(
+            content=File(
+                payload=payload,
+                file_name=f"file-${document_id}.pdf",
+                mime_type="application/pdf",
+            ),
+        ),
+    )
+    assert document
+    assert document.extracted_text is None
+    assert document.url is None
+    assert document.ingest_state == IngestState.PENDING
+    assert document.ingest_error is None
+
+
+async def document_ingested(client, document_id):
+    await asyncio.sleep(1)
+    status2 = await get_document_status.asyncio(document_id, client=client)
```

Comment:
This may need to be a loop to wait for the status to be non-pending. Especially when running the embedding model in CI, it may take more than 1 second. I've generally used some form of the following -- it means when things are fast, we don't waste 1 second running the test (especially important if we ingest multiple documents), and when things take longer we're ok:

```
status = await get_document_status.asyncio(document.id, client=client) 
while getattr(status, "ingest_state", IngestState.PENDING) == IngestState.PENDING:
    await asyncio.sleep(0.1)
    status = await get_document_status.asyncio(document.id, client=client) 
```